src/app/core/dspace-rest-v2/dspace-rest-v2.service.ts

```diff
@@ -45,9 +56,18 @@ export class DSpaceRESTv2Service {
    * @return {Observable<string>}
    *      An Observable<string> containing the response from the server
    */
-  request(method: RestRequestMethod, url: string, body?: any): Observable<DSpaceRESTV2Response> {
-    return this.http.request(method, url, { body, observe: 'response' })
-      .map((res) => ({ payload: res.body, statusCode: res.statusText }))
+  request(method: RestRequestMethod, url: string, body?: any, options?: HttpOptions): Observable<DSpaceRESTV2Response> {
+    const requestOptions: HttpOptions = {};
+    requestOptions.body = body;
+    requestOptions.observe = 'response';
+    if (options && options.headers) {
+      requestOptions.headers = Object.assign(new HttpHeaders(),  options.headers);
+    }
```

Comment:
Regarding the "disable ip check" problem. Based on this StackOverflow answer, it should be possible in the Angular app to check if you're running in a browser or in server-side rendering mode: https://stackoverflow.com/questions/44369945/get-remote-address-in-angular-universal#answer-45117793. Using an custom provider, you should also be able to retrieve the original HTTP request received by the express.js server (see the same StackOverflow answer). 

So can't we add some logic here like `if (isPlatformServer(this.platformId))` is true, then add an `X-Forwarded-For` header with the original client ip address from the Express request?

Reply:
(Note that I still need to learn a lot about Angular so I don't know if this is the best approach)

Reply:
That should be doable, but the same comment I made for platform.service.ts applies here as well: It would be better if we have a separate implementation for server and browser and load the correct version in the platform's main module file, instead of writing if server -> do this, if browser -> do that